SwiftGit2/Repository.swift

```diff
@@ -537,4 +537,149 @@ final public class Repository {
+				return self.tree(from: baseCommit)
+			}
+			.flatMap { baseTree in
+				guard !commit.parents.isEmpty else {
+					// Initial commit in a repository
+					return self.diff(withOldTree: nil, andNewTree: baseTree)
+				}
+
+				var mergeDiff: Result<OpaquePointer?, NSError> = .success(nil)
+				for parent in commit.parents {
+					mergeDiff = mergeDiff
+						.flatMap { mergeDiff in
+							return self
+								.parentCommit(from: parent)
+								.flatMap { commit in
+									return self.tree(from: commit)
+								}
+								.flatMap { tree in
+									return self.diff(withOldTree: tree, andNewTree: baseTree)
+								}
+								.flatMap { diff in
+									guard let mergeDiff = mergeDiff else { return .success(diff) }
+									let mergeResult = git_diff_merge(mergeDiff, diff)
+									guard mergeResult == GIT_OK.rawValue else {
+										return .failure(NSError(gitError: mergeResult, pointOfFailure: "git_diff_merge"))
+									}
+									return .success(mergeDiff)
+							}
+					}
+				}
+				return .success(mergeDiff.value!!)
+			}
+			.flatMap { diffResult in
+				return self.processDiffDeltas(diffResult)
+		}
+	}
+
+	private func commit(with oid: OID) -> Result<OpaquePointer, NSError> {
+		var unsafeBaseCommit: OpaquePointer? = nil
+		let unsafeBaseOid = UnsafeMutablePointer<git_oid>.allocate(capacity: 1)
+		git_oid_fromstr(unsafeBaseOid, oid.description)
+		let lookupBaseGitResult = git_commit_lookup(&unsafeBaseCommit, self.pointer, unsafeBaseOid)
+		guard lookupBaseGitResult == GIT_OK.rawValue, let unwrapBaseCommit = unsafeBaseCommit else {
+			return Result.failure(NSError(gitError: lookupBaseGitResult, pointOfFailure: "git_commit_lookup"))
+		}
+		git_commit_free(unsafeBaseCommit)
+
+		return Result.success(unwrapBaseCommit)
+	}
+
+	private func diff(withOldTree oldTree: OpaquePointer?,
+	                  andNewTree newTree: OpaquePointer?) -> Result<OpaquePointer, NSError> {
+		var unsafeDiff: OpaquePointer? = nil
+		let diffResult = git_diff_tree_to_tree(&unsafeDiff, self.pointer, oldTree, newTree, nil)
+		guard diffResult == GIT_OK.rawValue, let unwrapDiffResult = unsafeDiff else {
+			return Result.failure(NSError(gitError: diffResult, pointOfFailure: "git_diff_tree_to_tree"))
+		}
+
+		return Result.success(unwrapDiffResult)
+	}
+
+	private func parentCommit(from parent: PointerTo<Commit>) -> Result<OpaquePointer, NSError> {
+		var unsafeParentCommit: OpaquePointer? = nil
+		let unsafeParentOid = UnsafeMutablePointer<git_oid>.allocate(capacity: 1)
+		git_oid_fromstr(unsafeParentOid, parent.oid.description)
+		let lookupParentGitResult = git_commit_lookup(&unsafeParentCommit, self.pointer, unsafeParentOid)
+		guard lookupParentGitResult == GIT_OK.rawValue, let unwrapParentCommit = unsafeParentCommit else {
+			return Result.failure(NSError(gitError: lookupParentGitResult, pointOfFailure: "git_commit_lookup"))
+		}
+		git_commit_free(unsafeParentCommit)
+
+		return Result.success(unwrapParentCommit)
+	}
```

Comment:
This whole method should be unneeded since it basically repeats `commit(with: OID)` above.

Instead you should be able to do `commit(with: OID(parent.oid))`.

Reply:
Good catch :+1: